src/Sentry/Sentry.csproj

```diff
@@ -67,4 +68,8 @@
     <None Include="buildTransitive\Sentry.props" Pack="true" PackagePath="build\Sentry.props" />
   </ItemGroup>
 
+  <ItemGroup Condition="$(TargetFramework.StartsWith('net6.0'))">
+    <!-- TODO we only need LZ4 decompression only for Android, but we want to run all tests during development on net6.0/net7.0 directly on dev machine -->
+    <PackageReference Include="K4os.Compression.LZ4" Version="1.*" />
```

Comment:
This is a show stopper though. Can we add this dep only on the test project instead?

Reply:
I know, that's why there's the TODO above - this needs to be part of the Android build for runtime & also part of the test project - I don't know how to do that though... Open to suggestions how to make it work. When I tried adding to the test project, the Sentry project didn't compile because of the dependency. Likely, the whole AndroidAssemblyReader will need to move out of that main project. It's a bit unclear to me due to my limited knowledge of the solution structure and what integrates with what.

Reply:
You can move the package reference  to `/src/Sentry/Platforms/Android/Sentry.Android.props`.  It is already set up to be included only for the Android target.

`AndroidAssemblyReader.cs` should probably move under `/src/Sentry/Platforms/Android` also.  Everything in that folder is only included for Android targets.

Reply:
To be clear though - we are taking a true dependency on the LZ4 package at runtime for Android?  Is there another way?

Reply:
To clarify, I mean do we need the LZ4 decompression to actually occur in the user's shipping application? Or just at build time?  If the latter, then the package reference could go in `/src/Sentry/buildTransitive/Sentry.props`, with something like:

```xml
<ItemGroup Condition="$([MSBuild]::GetTargetPlatformIdentifier('$(TargetFramework)')) == 'android'">
    <PackageReference Include="K4os.Compression.LZ4" Version="1.*" PrivateAssets="All" />
</ItemGroup>
```

Reply:
If this goes into the `netX.0-android`-only target I think it's reasonable. We could go ahead

Reply:
Ok.  I will see if I can figure out if there's a way to access them in memory.  If we need to stay with the LZ4, then the place to add it is in `/src/Sentry/Platforms/Android/Sentry.Android.props`.  Just make a simple `<PackageReference>` with no extra tags, and do use an exact version rather than floating (we use floating versions in tests and samples only as a convenience to reduce update frequency, but they pose a security risk for distributable components.).

Reply:
I've resolved this by moving the LZ4 library and the `AndroidAssemblyReader` to compile for android only.  Separately the files are manually linked into a new `Sentry.Android.Tests` project that compile for regular .NET.